src/http.cc

```diff
@@ -1393,6 +1395,16 @@ HttpStateData::truncateVirginBody()
     }
 }
 
+/// called when an incomplete reply body preliminary ends with unexpected EOF
+void
+HttpStateData::failOnPrematureReplyBodyEof()
+{
+    const auto err = new ErrorState(ERR_READ_ERROR, Http::scBadGateway, fwd->request, fwd->al);
+    static const auto d = MakeNamedErrorDetail("SRV_PREMATURE_EOF");
+    err->detailError(d);
+    fwd->fail(err);
+}
+
```

Comment:
Please post a before/after diff of access.log records (with just the key fields) for Daft truncated-responses (and any other relevant) test. We should show what we fixed (and what information is present in fixed records).

Reply:
I extracted the relevant access.log fields and adjusted with them the PR description.